addons/microsoft_calendar/models/calendar.py

```diff
@@ -94,6 +93,29 @@ def _microsoft_to_odoo_values(self, microsoft_event, default_reminders=(), defau
             'recurrency': microsoft_event.is_recurrent()
         }
 
+        # This is the tricky part of this improvement.
+        # It totally bypasses the logic of using extended property to store real event owner.
+        # If the sync Microsoft account is the organizer for the Microsoft event:
+        #   Case 1: the sync Odoo user should be the real owner of the event
+        #       if the event is a new created in Microsoft Calendar, and there isn't corresponding Odoo event:
+        #           We do nothing to value["user_id"].
+        #           Odoo will create an Odoo event without specifying "user_id".
+        #           The sync Odoo user(creator) will become the event owner by default.
+        #       if the event has a corresponding Odoo event:
+        #           We do nothing to value["user_id"].
+        #           The owner of the Odoo will not be changed.
+        #   Case 2: Originally, this is a shared event owned by another Odoo user in Odoo Calendar. Later, it is synced to and updated in Microsoft Calendar
+        #       We do nothing to value["user_id"].
+        #       The real owner of the Odoo event will not be changed
+        # Else: (the sync Microsoft account is not the organizer for the Microsoft event)
+        #   Case 1: Originally, it is a Microsoft event owned by another Microsoft account and shared to this sync Microsoft account
+        #       We make the sync Odoo user become the owner of the event in Odoo Calendar
+        #       He can modify this event locally. However, the modified info will not be updated in Microsoft Calendar.
+        #       And if there is any updates in its Microsoft event, all modified info will be refreshed.
+        # As a result, we don't need any extended property in Microsoft event to store the real owner of the Odoo event.
+        if not microsoft_event.isOrganizer:
```

Comment:
Looks good to me except when the event:
- has been created in Outlook,
- the event is synced by a Odoo user who is an attendee of this event,
- the event organizer exists in Odoo.
In this case, as it was done before, I think we should find the organizer using the email set in the Outlook event.

So, as discussed on Discord, we could do something like:
1) if we don't want to allow an attendee modifying the event from Odoo when the organizer is not a Odoo user:
```python
if not microsoft_event.isOrganizer:
    values["user_id"] = user_id.id if env['res.users'].search([('email', '=', self.organizer['emailAddress']['address'])], limit=1) else False
```
2) if we want to allow an attendee to modify the Outlook event:
```python
if not microsoft_event.isOrganizer:
    values["user_id"] = user_id.id if env['res.users'].search([('email', '=', self.organizer['emailAddress']['address'])], limit=1) else self.env.user.id
```